security/pkg/util/k8sutil_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"encoding/json"
+	"encoding/pem"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"testing"
+
+	k8sauth "k8s.io/api/authentication/v1"
+)
+
+type mockAPIServer struct {
+	httpServer *httptest.Server
+	apiPath    string
+	token      string
+}
+
+type clientConfig struct {
+	tlsCert     []byte
+	reviewPath  string
+	clientToken string
+}
+
+func TestOnMockAPIServer(t *testing.T) {
+	testCases := map[string]struct {
+		cliConfig    clientConfig
+		expectedCert []string
+		expectedErr  string
+	}{
+		"Valid request": {
+			cliConfig:   clientConfig{tlsCert: []byte{}, reviewPath: "review-path", clientToken: "Bearer fake-client-token"},
+			expectedErr: "",
+		},
+		"Wrong review path": {
+			cliConfig:    clientConfig{tlsCert: []byte{}, reviewPath: "wrong-review-path", clientToken: "Bearer fake-client-token"},
+			expectedCert: nil,
+			expectedErr:  "the service account authentication returns an error",
+		},
+		"Wrong client token": {
+			cliConfig:    clientConfig{tlsCert: []byte{}, reviewPath: "review-path", clientToken: "wrong-client-token"},
+			expectedCert: nil,
+			expectedErr:  "the service account authentication returns an error",
+		},
```

Comment:
Can you add more tests? See: https://github.com/istio/istio/pull/10657/files#diff-92e06272650a4cc4bc37773ac29289ce

Reply:
Done: added 4 new tests.